src/librbd/crypto/luks/FormatRequest.h

```diff
@@ -0,0 +1,56 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CRYPTO_LUKS_FORMAT_REQUEST_H
+#define CEPH_LIBRBD_CRYPTO_LUKS_FORMAT_REQUEST_H
+
+#include "librbd/ImageCtx.h"
+#include "librbd/crypto/Types.h"
+#include "librbd/crypto/luks/Header.h"
+
+namespace librbd {
+
+class ImageCtx;
+
+namespace crypto {
+namespace luks {
+
+template <typename I>
+class FormatRequest {
+public:
+    static FormatRequest* create(
+            I* image_ctx, DiskEncryptionFormat type, CipherAlgorithm cipher,
+            const char* passphrase, size_t passphrase_size, Context* on_finish,
```

Comment:
Nit: can we just internally use an `std::string` or `ceph::bufferlist` instead of having to pass a pointer and length and manage memory?

Reply:
Problem is I want to do a best effort for wiping.
Also did the same for the openssl::DataCryptor.
Should I use std::string, and try one of the solutions suggested here?
https://stackoverflow.com/questions/38702943/how-to-cleanse-overwrite-with-random-bytes-stdstring-internal-buffer/38703073

Reply:
You could use a string w/ move semantics and then just flush it via the explicit zero op at destruction.